Update Calendar block icon for better alignment with the Archives block icon

kjellr opened the pull request:
The current Calendar block icon uses the Calendar Dashicon, which is stylistically different from the Calendar used in the Archives block. To better align these, I suggest we adopt the [`calendar_today` icon from Material](https://material.io/tools/icons/?search=cal&icon=calendar_today&style=outline) to represent the calendar block. 

Alternatively, to separate these two icons visually, we could use the current "Archives" icon for the Calendar block instead, and explore a new icon for the Archives block — Perhaps the [`history`](https://material.io/tools/icons/?icon=history&style=outline), [`book`](https://material.io/tools/icons/?search=book&icon=book&style=outline), or [`library_book`](https://material.io/tools/icons/?search=book&icon=library_books&style=outline) icons?

---

_Before:_ 

![Screen Shot 2019-05-14 at 9 32 47 AM](https://user-images.githubusercontent.com/1202812/57702203-8e9f8100-762b-11e9-8931-723809ab84d1.png)


_After:_ 

![Screen Shot 2019-05-14 at 9 30 24 AM](https://user-images.githubusercontent.com/1202812/57702208-9101db00-762b-11e9-8113-0258f45259d3.png)

kjellr committed:
Update Calendar block icon to match Archives block

Soean reviewed (approved):
Code looks good and works in inserter and block.

Soean commented on packages/block-library/src/calendar/icon.js:
```diff
@@ -0,0 +1,8 @@
+import { Path, SVG } from '@wordpress/components';
+
+export default (
+	<SVG width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg"><Path d="M21 6V20C21 21.1 20.1 22 19 22H5C3.89 22 3 21.1 3 20L3.01 6C3.01 4.9 3.89 4 5 4H6V2H8V4H16V2H18V4H19C20.1 4 21 4.9 21 6ZM5 8H19V6H5V8ZM19 20V10H5V20H19Z" /></SVG>
```
Minor: We can remove the `fill="none"`, because it should have a color and its overridden by CSS.
```suggestion
	<SVG width="24" height="24" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><Path d="M21 6V20C21 21.1 20.1 22 19 22H5C3.89 22 3 21.1 3 20L3.01 6C3.01 4.9 3.89 4 5 4H6V2H8V4H16V2H18V4H19C20.1 4 21 4.9 21 6ZM5 8H19V6H5V8ZM19 20V10H5V20H19Z" /></SVG>
```

kjellr replied:
👍 Good catch, thanks. Updated in 63b2db9

kjellr committed:
Remove unnecessary fill=none

sarahmonster commented:
From a design perspective, this definitely feels like an improvement! 👍 

For reference, this is the Calendar block:
<img width="598" alt="Screenshot 2019-05-14 at 20 40 12" src="https://user-images.githubusercontent.com/376315/57727098-7fc5c800-7645-11e9-8ed9-5567ad7938ce.png">

...and this is the Archive block:
<img width="604" alt="Screenshot 2019-05-14 at 20 40 21" src="https://user-images.githubusercontent.com/376315/57727109-86ecd600-7645-11e9-8d01-a30b67eeea73.png">

Given the obvious visual difference betweeen these two blocks, it might be a good idea to consider representing that difference in the icons we use. The Archive icon seems to be a better representation of the Calendar block here, in terms of visual appearance and output. From the existing Material library, the `library_book` definitely feels like the _closest_ choice, but it still doesn't immediately feel as though it fits.

What about something like [list_alt](https://material.io/tools/icons/?search=list&icon=list_alt&style=outline)? Or maybe a list inside a calendar, to communicate the idea of a list of past posts?

melchoyce commented:
> From the existing Material library, the library_book definitely feels like the closest choice, but it still doesn't immediately feel as though it fits.

I thought the same, it felt closest of the bunch.

There's a lot of filing boxes here: https://www.iconfinder.com/search/?q=archive

We could create a custom icon, maybe like this:

![image](https://user-images.githubusercontent.com/2846578/57728354-bdd0e500-7661-11e9-92f5-6c8dc824dbe7.png)

kjellr commented:
> What about something like `list_alt`? Or maybe a list inside a calendar, to communicate the idea of a list of past posts?

`list-alt` almost makes sense, but I know we're actually considering using it for [another control in the UI](https://github.com/WordPress/gutenberg/issues/15096#issuecomment-486235798). 😄 

The list inside of a calendar may work:

<img width="689" alt="Screen Shot 2019-05-14 at 4 25 37 PM" src="https://user-images.githubusercontent.com/1202812/57729763-f920e300-7664-11e9-99f6-bf02f19d737f.png">

@melchoyce the middle one's just like your example above, but I added back the tabs at the top so it looks less like a web browser.

drw158 commented:
Archive:

example from Gridicons

![Screen Shot 2019-05-14 at 3 17 17 PM](https://user-images.githubusercontent.com/618551/57729301-78111e00-765b-11e9-8d84-595a39f91bd2.png)

Ones that might work from Material

<img width="117" alt="Screen Shot 2019-05-14 at 2 40 24 PM" src="https://user-images.githubusercontent.com/618551/57729315-819a8600-765b-11e9-9393-335ec89f68e3.png">
![Screen Shot 2019-05-14 at 3 23 03 PM](https://user-images.githubusercontent.com/618551/57729849-c83cb000-765c-11e9-98c2-fdbb819e3aa1.png)
![Screen Shot 2019-05-14 at 3 22 22 PM](https://user-images.githubusercontent.com/618551/57729850-c83cb000-765c-11e9-9e74-5bc12c051a21.png)
![Screen Shot 2019-05-14 at 3 21 24 PM](https://user-images.githubusercontent.com/618551/57729851-c83cb000-765c-11e9-8348-10f376542e93.png)
![Screen Shot 2019-05-14 at 3 21 21 PM](https://user-images.githubusercontent.com/618551/57729852-c83cb000-765c-11e9-9341-f4c8d73abe17.png)
![Screen Shot 2019-05-14 at 3 20 50 PM](https://user-images.githubusercontent.com/618551/57729853-c83cb000-765c-11e9-8129-7f155172bfc1.png)
![Screen Shot 2019-05-14 at 3 20 44 PM](https://user-images.githubusercontent.com/618551/57729854-c83cb000-765c-11e9-95bf-6a7ad87531e8.png)

kjellr committed:
Update icons

Use the old Archives icon for the Calendar block, add a new icon to the Archives block.

kjellr commented:
Thanks for the feedback folks. I've updated the PR like so: 

- I've updated the Calendar icon to use the old Archives block icon.
- I've added a new icon to the Archives block. I went with the "List inside of a calendar" option for now, but happy to refresh with one of the other options discussed. This one made sense to me right now: the archives block is basically a calendar, but displayed in a list form. This icon helps both blocks share the same visual language, which is kind of cool. Maybe they're _too_ similar though? 

![Screen Shot 2019-05-15 at 12 37 47 PM](https://user-images.githubusercontent.com/1202812/57793501-75700080-770f-11e9-9156-675f2077c09c.png)

Let me know what you all think!

mapk commented:
:shipit:  Looks good Kjell!

mapk reviewed (approved):
:shipit: Dropped a proper review. 😉

kjellr commented:
Thanks, folks!
